src/main/java/com/example/ecommerce/controller/ProductController.java

```diff
@@ -26,18 +26,76 @@ public class ProductController {
 	private CategoryService categoryService;
 
 	@GetMapping("categories/{categoryId}/products")
-	public String index(@PathVariable(name = "categoryId") Integer categoryId, Model model,
-			HttpServletResponse response) throws IOException {
-		
+	public String index(@PathVariable(name = "categoryId") Integer categoryId,
+			@RequestParam(value = "page", defaultValue = "1") Integer page, Model model, HttpServletResponse response)
+			throws IOException {
+
+		int pageSize = 8;
+		Integer currentPage;
+		if (page != null) {
+			currentPage = page;
+		} else {
+			currentPage = 1;
+		}
+		List<Product> products;
+		int startIndex = (currentPage - 1) * pageSize;
 		if (categoryId == null) {
 			response.sendRedirect("/404-page");
 			String errorMessage = "Không tìm thấy sản phẩm";
 			model.addAttribute("errorMessage", errorMessage);
 			return "error";
 		}
-		List<Product> products = this.productService.getProductsByCategory(categoryId);
-		model.addAttribute("products", products);
+		products = this.productService.getProductsByCategory(categoryId);
+		int endIndex = Math.min(startIndex + pageSize, products.size());
+		List<Product> productsOnPage = products.subList(startIndex, endIndex);
+		model.addAttribute("products", productsOnPage);
+		model.addAttribute("currentPage", currentPage);
+		model.addAttribute("totalPages", (int) Math.ceil((double) products.size() / pageSize));
 		return "products/index";
+
+	}
+
+	@GetMapping(value = "categories/{categoryId}/products", params = { "page", "sortBy" })
+	public String sort(@PathVariable(name = "categoryId") Integer categoryId,
+			@RequestParam(value = "page", defaultValue = "1") Integer page,
+			@RequestParam(value = "sortBy") String sortBy, 
+			@RequestParam(value = "ord", required = false) String order,
+			Model model, HttpServletResponse response) throws IOException {
+
+		int pageSize = 8;
+		Integer currentPage;
+		if (page != null) {
+			currentPage = page;
+		} else {
+			currentPage = 1;
+		}
+		List<Product> products = null;
+		int startIndex = (currentPage - 1) * pageSize;
+		if (sortBy.equals("sales")) {
+			products = this.productService.sortByBuyCount(categoryId);
+			int endIndex = Math.min(startIndex + pageSize, products.size());
+			List<Product> productsOnPage = products.subList(startIndex, endIndex);
+			model.addAttribute("products", productsOnPage);
+			model.addAttribute("currentPage", currentPage);
+			model.addAttribute("totalPages", (int) Math.ceil((double) products.size() / pageSize));
+			return "products/sort/sales";
+		} else if (sortBy.equals("price")) {
+			if (order != null && !order.isEmpty()) {
+				if (order.equals("ASC")) {
+					products = this.productService.sortByPrice(categoryId, "ASC");
+				} else if (order.equals("DESC")) {
+					products = this.productService.sortByPrice(categoryId, "DESC");
+				}
```

Comment:
```suggestion
				products = this.productService.sortByPrice(categoryId, order);
```
sao ko nhận param lun ma phải ri